test/effects.js

```diff
@@ -83,3 +83,25 @@ tests['test 3xx portamento'] = function(assert) {
   assert.equal(ch.period, 1152 - 16, 'row 1 tick 2 period -16');
 };
 
+tests['test Gxx global volume'] = function(assert) {
+  var xm = testdata.resetXMData();
+  // [pat][row][channel]
+  // 1 channel, 3 row blank pattern
+  xm.patterns = [
+    [
+      [[48, 1, -1, 16, 0x40]], // C-4  1 -- G40
+      [[48, 1, -1, 16, 0x2B]], // C-4  1 -- G2B
+      // test out of bounds volume
+      [[48, 1, -1, 16, 0x80]]  // C-4  1 -- G80
+    ]
+  ];
+  XMPlayer.xm.tempo = 1;
+  XMPlayer.nextTick();
+  // volume gets multiplied by 2 to match
+  // the initial max global volume of 128
+  assert.equal(XMPlayer.xm.global_volume, 0x40*2, 'global volume set to 0x40');
+  XMPlayer.nextTick();
+  assert.equal(XMPlayer.xm.global_volume, 0x2B*2, 'global volume set to 0x2B');
+  XMPlayer.nextTick();
+  assert.equal(XMPlayer.xm.global_volume, 0x40*2, 'global volume set to 0x40');
```

Comment:
wait, is it actually 64? oh, i see, the command is doubled. sorry, didn't realize that.